src/v/iceberg/snapshot_json.cc

```diff
@@ -0,0 +1,133 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#include "iceberg/snapshot_json.h"
+
+#include "container/chunked_hash_map.h"
+#include "iceberg/json_utils.h"
+#include "iceberg/snapshot.h"
+#include "json/document.h"
+#include "model/timestamp.h"
+#include "strings/string_switch.h"
+
+namespace iceberg {
+
+namespace {
+
+snapshot_operation operation_from_str(const ss::sstring& operation_str) {
+    return string_switch<snapshot_operation>(operation_str)
+      .match("append", snapshot_operation::append)
```

Comment:
nit: the neat thing i've been seeing is to make `operation_to_str` (below) return a `constexpr std::string_view`, and then here use `.match(operation_to_string(snapshot_operation::append), snapshot_operation::append)`.

Reply:
Done